client/web/src/components/externalServices/ExternalServiceEditPage.tsx

```diff
@@ -94,15 +94,61 @@ export const ExternalServiceEditPage: FC<Props> = ({
 
     const path = useMemo(() => getBreadCrumbs(externalService, true), [externalService])
 
-    const externalServiceCategory = resolveExternalServiceCategory(externalService, ghApp)
-
     const combinedError = fetchError || updateExternalServiceError || fetchGHAppError
     const combinedLoading = fetchLoading || updateExternalServiceLoading || fetchGHAppLoading
 
     if (updated && !combinedLoading && externalService?.warning === null) {
         navigate(`/site-admin/external-services/${encodeURIComponent(externalService.id)}`, { replace: true })
     }
 
+    const renderService = (externalService: ExternalServiceFieldsWithConfig): JSX.Element => {
+        const externalServiceCategory = resolveExternalServiceCategory(externalService, ghApp)
```

Comment:
We should use memo here as well to avoid unnecessary computation
```suggestion
        const externalServiceCategory = useMemo(() => resolveExternalServiceCategory(externalService, ghApp), [externalService, ghApp])
```